src/Service/InitializeAppData.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\PimApi\Model\Catalog;
+use App\PimApi\PimCatalogApiClient;
+use App\Storage\CatalogIdStorageInterface;
+
+final class InitializeAppData
+{
+    public function __construct(
+        private readonly CatalogIdStorageInterface $catalogIdStorage,
+        private readonly PimCatalogApiClient $pimCatalogApiClient,
+    ) {
+    }
+
+    public function __invoke(): void
+    {
+        $catalogs = $this->pimCatalogApiClient->getCatalogs();
+        $valueFilterCatalog = $this->findCatalogWithName($catalogs, Catalog::PRODUCT_VALUE_FILTERS_NAME);
+        $attributeMappingCatalog = $this->findCatalogWithName($catalogs, Catalog::ATTRIBUTE_MAPPING_NAME);
+
+        if (null === $valueFilterCatalog) {
+            $valueFilterCatalog = $this->pimCatalogApiClient->createCatalog(Catalog::PRODUCT_VALUE_FILTERS_NAME);
+        }
+
+        if (null === $attributeMappingCatalog) {
+            $attributeMappingCatalog = $this->pimCatalogApiClient->createCatalog(Catalog::ATTRIBUTE_MAPPING_NAME);
+            $this->pimCatalogApiClient->setProductMappingSchema($attributeMappingCatalog->id, $this->getProductMappingSchema());
```

Comment:
InitializeAppData is invoked on each callback, even when a user opens the app on the PIM side thus it would needlessly update the schema. Would you still move it out?